web/src/js/ducks/utils/list.js

```diff
@@ -1,166 +1,91 @@
-import {fetchApi} from "../../utils"
-
-export const ADD = "ADD"
-export const UPDATE = "UPDATE"
-export const REMOVE = "REMOVE"
-export const REQUEST_LIST = "REQUEST_LIST"
-export const RECEIVE_LIST = "RECEIVE_LIST"
+import _ from 'lodash'
 
+export const SET = 'LIST_SET'
+export const CLEAR = 'LIST_CLEAR'
+export const REQUEST = 'LIST_REQUEST'
+export const RECEIVE = 'LIST_RECEIVE'
 
 const defaultState = {
-    list: [],
-    isFetching: false,
-    actionsDuringFetch: [],
-    byId: {},
-    indexOf: {},
+    data: {},
+    pendingActions: null,
 }
 
-export default function makeList(actionType, fetchURL) {
-    function reduceList(state = defaultState, action = {}) {
-
-        if (action.type !== actionType) {
-            return state
-        }
+export default function reduce(state = defaultState, action) {
+    switch (action.type) {
 
-        // Handle cases where we finished fetching or are still fetching.
-        if (action.cmd === RECEIVE_LIST) {
-            let s = {
-                isFetching: false,
-                actionsDuringFetch: [],
-                list: action.list,
-                byId: {},
-                indexOf: {}
-            }
-            for (let i = 0; i < action.list.length; i++) {
-                let item = action.list[i]
-                s.byId[item.id] = item
-                s.indexOf[item.id] = i
-            }
-            for (action of state.actionsDuringFetch) {
-                s = reduceList(s, action)
+        case SET:
```

Comment:
Why do we have `SET` and `CLEAR` here now? What is `list` supposed to be doing now?